doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+4. VRF VNI Mapping configuration
+   - switch(config-if-vtep1) [no] map vrf VRF-Blue vni 10001
+5. Neighbor suppression
+   - switch(config) interface Vlan <vlan-id-xxx>
+   - switch(conf-if-vlanxxx)# [no] neigh-suppress
+   - This command will suppress both ARP & ND flooding over the tunnel when Vlan is extended.
+
+```
+
+#### 5.2.2 Show Commands
+
+     The show command outputs shall be from the exec mode of the KLISH shell and has the same
+     content as the show commands in the Click show section. 
+     In addition the following commands will be supported.
+
+```
+1. show vlan id and brief will be enhanced to display DIP. (tunnel members from  APP DB)
+
+2. show mac /show mac -v will be enhanced for EVPN MACs.
+   - Port column will display DIP. (ASIC DB view)
+   - Type column will display EVPN_static/EVPN_dynamic
+
+3. show NeighbourSuppressStatus to display Neighbor suppression status.
+
+```
+
+#### 5.2.3 Validations 
+
+```
+- VLAN to be configured before map vlan vni command is executed. 
+- VLAN cannot be deleted if there is a vlan vni map. 
+- source to be configured before  map vlan vni command is executed.
+- source cannot be removed if there are mappings already present. 
+- interface vxlan cannot be deleted if evpn nvo is configured. 
+- VRF is already configured    
+- VNI VLAN map should be configured prior to VRF VNI map configuration, since VNI should be both L2 and L3.
+- VNI VLAN map cannot be deleted if there is corresponding VRF VNI map.
+
+```
+
+## 6 Serviceability and Debug
+
+The existing logging mechanisms shall be used. The show commands and the redis table
+dumps can be used as debug aids.
+
+## 7 Warm Reboot Support
+
+Warm reboot will support below requirements
+
+1. System Warm reboot for EVPN 
+   The goal of the system warm reboot is to achieve system software reload without impacting the data plane and graceful restart of the control plane (BGP).
+
+![System Warm reboot and reconciliation sequence](images/warmrebootseq.PNG "Figure 16: System Warm reboot and reconciliation sequence")
+
+__Figure 16: System Warm reboot and reconciliation sequence__
+
+During the system warm-reboot restoration/reconciliation, the fdbsyncd is dependent on Intf, Vlan, Vrf, Vxlan Mgrs to complete their data restoration to kernel. Before fdbsyncd can program the MACs to the kernel it needs to wait for these subsystems data replay to kernel. Without this the kernel programming for fdbsyncd will fail.  Subsequently if the FDB/VTEP info is not available in kernel/zebra for BGP to distribute to its peers, it will converge with premature info causing data-loss. To solve this dependency , we use a replay-done notification by each subsystem. The notification is passed between the subsystems by using the existing  WARM_RESTART_TABLE. A new state "replayed" in the data restoration flow  ( "state" = "replayed") is used by each subsystem to indicate it has replayed its data to kernel.
```

Comment:
>A new state "replayed" [](start = 666, length = 22)

The state enum becomes  more complex. Could you draw a state transition diagram and describe it?

Reply:
will add more info there